pkg/syncer/fullsync.go

```diff
@@ -191,16 +209,27 @@ func fullSyncDeleteVolumes(ctx context.Context, volumeIDDeleteArray []cnstypes.C
 		log.Errorf("FullSync: fullSyncDeleteVolumes failed to get PVs from kubernetes. Err: %v", err)
 		return
 	}
+
+	var volumeMigrationService migration.VolumeMigrationService
+	// Instantiate volumeMigrationService when migration feature state is True
+	if migrationFeatureStateForFullSync {
+		volumeMigrationService, err = migration.GetVolumeMigrationService(ctx, &metadataSyncer.volumeManager, metadataSyncer.configInfo.Cfg)
+		if err != nil {
+			log.Errorf("FullSync: Failed to get migration service. Err: %v", err)
+			return
+		}
+	}
 	// Create map for easy lookup
 	for _, pv := range currentK8sPV {
 		var volumeHandle string
-		var err error
-		if migrationFeatureStateForFullSync && pv.Spec.VsphereVolume != nil {
-			migrationVolumeSpec := migration.VolumeSpec{VolumePath: pv.Spec.VsphereVolume.VolumePath, StoragePolicyName: pv.Spec.VsphereVolume.StoragePolicyName}
-			volumeHandle, err = volumeMigrationService.GetVolumeID(ctx, migrationVolumeSpec)
-			if err != nil {
-				log.Errorf("FullSync: Failed to get VolumeID from volumeMigrationService for migration VolumeSpec: %v with error %+v", migrationVolumeSpec, err)
-				return
+		if pv.Spec.VsphereVolume != nil {
```

Comment:
Can we merge nested if conditions`if pv.Spec.VsphereVolume != nil && migrationFeatureStateForFullSync`

Reply:
No. migrationFeatureStateForFullSync can be `False`
<pre>
if pv.Spec.VsphereVolume != nil && migrationFeatureStateForFullSync {
}
else {
// Does not mean it is always CSI volume 
}



if pv.Spec.VsphereVolume != nil {
      if migrationFeatureStateForFullSync {
       }
}
else {
// Always CSI volume
}
</pre>

Reply:
Can you add this as comment in the code?

Reply:
@chethanv28 That example is not entirely correct. `currentK8sPVs` is formed from `getPVsInBoundAvailableOrReleased`, which filters to only add PVs that are either vSphere CSI or vSphere in-tree volumes.

Reply:
Done